### PR TITLE
sidecar: implement without replica labels capability

### PR DESCRIPTION
**Pull request by GiedriusS**
Implement the new without replica labels capability in Sidecar to have more efficient querying.

### CODE REVIEW

File: pkg/store/prometheus.go [hunk 1]
```diff
@@ -330,7 +336,7 @@ func (p *PrometheusStore) handleSampledPrometheusResponse(
 	span.SetTag("series_count", len(resp.Results[0].Timeseries))
 
 	for _, e := range resp.Results[0].Timeseries {
-		lset := labelpb.ExtendSortedLabels(labelpb.ZLabelsToPromLabels(e.Labels), extLset)
+		lset := labelpb.ExtendSortedLabels(labelpb.ZLabelsToPromLabels(e.Labels), rmLabels(extLset.Copy(), extLsetToRemove))
```
Comment (GiedriusS) [inline pkg/store/prometheus.go:339]: Copy is needed because replica labels could be set per-request :cry:

Comment (fpetkovski) [reply pkg/store/prometheus.go:339]: I think this is fine, it's only copying the slice without making a deep copy.

File: pkg/store/prometheus.go [hunk 2]
```diff
@@ -198,8 +198,13 @@ func (p *PrometheusStore) Series(r *storepb.SeriesRequest, s storepb.Store_Serie
 	shardMatcher := r.ShardInfo.Matcher(&p.buffers)
 	defer shardMatcher.Close()
 
+	extLsetToRemove := map[string]struct{}{}
+	for _, lbl := range r.WithoutReplicaLabels {
+		extLsetToRemove[lbl] = struct{}{}
+	}
```
Comment (fpetkovski) [inline pkg/store/prometheus.go:204]: I know this is still draft, but should we remove the labels from the existing `extLset` here? This way everything else will keep working and we don't need to do `rmLabel` everywhere else.

Comment (GiedriusS) [reply pkg/store/prometheus.go]: @fpetkovski you are correct, I had to move this chunk a bit to cover `shipChunks` case. I actually noticed that Prometheus already sends labels with external labels in them so I'm not sure why we were always calling `labels.Extend`. I have removed that.

Comment (fpetkovski) [reply pkg/store/prometheus.go]: Makes sense. The good thing with this is that we don't need to copy labels.

**Review by fpetkovski — APPROVED** [review]
Thanks, LGTM 👍

**Pull request merged**